### PR TITLE
Apache load balancer chart initial version

### PR DESCRIPTION
**Pull request by simon-meng-cn**

### CODE REVIEW

**Comment by moreaut** [comment]
Overall, this helm chart looks nothing like the operator helm chart.  It probably should.

The operator helm chart has much more validation of input parameters to catch user errors.

**Comment by rjeberhard** [comment]
@simon-meng-cn @moreaut It's hard for me to tell which comments have been resolved.  Can you please give me an idea of how close we are on this PR?

**Comment by moreaut** [comment]
I don't think any of the issues I raised have been addressed.
I don't think we're close if you want this helm chart to look like the operator helm chart.
We could put it in as is, then fix it later, but then anyone using would have to catch up.

**Comment by lilyhe123** [comment]
This chart need to be used in java integration tests to make the apache tests pass. This can be done in this change or separate change. Both are fine to me.

File: kubernetes/samples/charts/apache-webtier/templates/cluster-role-binding.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{{ if .Values.rbac.create }}
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: {{ template "apache.fullname" . }}
+  labels:
+    app: {{ template "apache.fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
```
[the remark this answers is not in the document]
Comment (simon-meng-cn) [reply kubernetes/samples/charts/apache-webtier/templates/cluster-role-binding.yaml]: My understanding is that the install chart preferably does not contain a domain related resources.
If it contains known domainUID or domainName, how to process the new added domains?

**Comment by rjeberhard** [comment]
I'm going to go ahead with merging this PR and we can address Tom's concern that the charts should be similar going forward.

File: kubernetes/samples/charts/apache-webtier/README.md [hunk 2]
```diff
@@ -54,6 +54,7 @@ The following table lists the configurable parameters of the Apache webtier char
 | `volumePath`                       | Docker volume path for apache webtier                         | ``                    |
 | `createRBAC`                       | Boolean indicating if RBAC resources should be created        | `true`                |
 | `httpNodePort`                     | NodePort to expose for http access                            | `30305`               |
+| `httpsNodePort`                    | NodePort to expose for https access                           | `30443`               |
```
Comment (doxiao) [inline kubernetes/samples/charts/apache-webtier/README.md]: How do we handle certificate files for SSL?

File: kubernetes/samples/charts/apache-webtier/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+apiVersion: v1
```
Comment (doxiao) [inline kubernetes/samples/charts/apache-webtier/Chart.yaml]: Add Copyright?

File: kubernetes/samples/charts/apache-webtier/templates/cluster-role-binding.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{{ if .Values.createRBAC }}
```
Comment (doxiao) [inline kubernetes/samples/charts/apache-webtier/templates/cluster-role-binding.yaml]: Add Copyright?

File: kubernetes/samples/charts/apache-webtier/templates/cluster-role.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+{{ if .Values.createRBAC }}
```
Comment (doxiao) [inline kubernetes/samples/charts/apache-webtier/templates/cluster-role.yaml]: Copyright?

File: kubernetes/samples/charts/apache-webtier/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
+kind: Deployment
```
Comment (doxiao) [inline kubernetes/samples/charts/apache-webtier/templates/deployment.yaml]: Copyright?

File: kubernetes/samples/charts/apache-webtier/templates/service-account.yaml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
```
Comment (doxiao) [inline kubernetes/samples/charts/apache-webtier/templates/service-account.yaml]: Copyright?

File: kubernetes/samples/charts/apache-webtier/templates/service.yaml [hunk 8]
```diff
@@ -0,0 +1,21 @@
+apiVersion: v1
```
Comment (doxiao) [inline kubernetes/samples/charts/apache-webtier/templates/service.yaml]: Copyright

File: kubernetes/samples/charts/apache-webtier/README.md (hunk 2 above)
Comment (simon-meng-cn) [reply kubernetes/samples/charts/apache-webtier/README.md]: The default certificate files are in apache image. We can use volume to provide custom certificate files. Then edit apache ssl configuration file to reference the custom certificate files.

File: kubernetes/samples/charts/apache-webtier/Chart.yaml (hunk 3 above)
Comment (simon-meng-cn) [reply kubernetes/samples/charts/apache-webtier/Chart.yaml]: I will submit another change to add copyright.

**Pull request merged**